app/src/main/java/com/aws/amazonlocation/ui/main/explore/ExploreViewModel.kt

```diff
@@ -365,6 +403,15 @@ class ExploreViewModel
                             .summary
                             ?.distance
                             ?.toDouble()
+                    val getLastTime =
+                        if (legs.last().vehicleLegDetails != null) {
+                            legs.last().vehicleLegDetails?.arrival?.time
+                        } else if (legs.last().pedestrianLegDetails != null) {
+                            legs.last().pedestrianLegDetails?.arrival?.time
+                        } else if (legs.last().ferryLegDetails != null) {
+                            legs.last().ferryLegDetails?.arrival?.time
+                        } else ""
```

Comment:
This should never happen, and if it does it's a service side bug. Can we throw an error instead?